File: backend/app/api/routers/optimization.py
from fastapi import APIRouter
from ...services.optimizer_service import run_optimization, run_best_choice
from ...schemas.requests import OptimizationRequest, BestChoiceRequest
from ...schemas.responses import OptimizationResponse, BestChoiceResponse

router = APIRouter()

@router.post("/run", response_model=OptimizationResponse)
def run(request: OptimizationRequest):
    return run_optimization(request)

@router.post("/best_choice", response_model=BestChoiceResponse)
def best_choice(request: BestChoiceRequest):
    row = request.model_dump()
    return run_best_choice(row)

File: src/api/optimization.ts
import type { OptimizationRequest, OptimizationResponse, BestChoiceRequest, BestChoiceResponse } from "../types/api";
import { apiFetch } from "./client";

export function runOptimization(req: OptimizationRequest): Promise<OptimizationResponse> {
  return apiFetch<OptimizationResponse>("/optimization/run", {
    method: "POST",
    body: JSON.stringify(req),
  });
}

export function bestChoice(req: BestChoiceRequest): Promise<BestChoiceResponse> {
  return apiFetch<BestChoiceResponse>("/optimization/best_choice", {
    method: "POST",
    body: JSON.stringify(req),
  });
}
